reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py

```diff
@@ -102,6 +104,20 @@ def _assemble_subscribers_with_auto_promotions(self) -> None:
                     soak_days = (
                         target.promotion.soak_days if target.promotion.soak_days else 0
                     )
+                    schedule = (
+                        target.promotion.schedule
+                        if target.promotion.schedule
+                        else "* * * * *"
+                    )
+                    # Ideally we would catch that at schema validation time
+                    if not croniter.is_valid(schedule):
+                        logging.error(
+                            "Subscriber at %s has an invalid schedule declaration %s. We will block any promotion for that target until this is fixed.",
+                            file_path,
+                            schedule,
+                        )
+                        # Lets set schedule for impossible date to block promotions
```

Comment:
Why not raise an exception here?

Reply:
Good point. I moved the validation logic now to the subscriber itself - makes this a lot easier.